actions/incremental-rollout-appservice/action.yml

```
@@ -0,0 +1,122 @@
+name: "Incremental Rollout for App Service"
+description: "Incrementally shift traffic to staging slot, run canary script, shift traffic and finalize with slot swap."
+
+inputs:
+  resource_group_name:
+    description: "Azure Resource Group name"
+    required: true
+  web_app_name:
+    description: "Azure Web App name"
+    required: true
+  script_language:
+    description: "Script language (typescript (npm) or typescript (yarn))"
+    required: true
+    default: "typescript (npm)"
+  script_name:
+    description: "Script name to execute during canary deployment"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Incremental Rollout to Staging Slot
+      shell: bash
+      env:
+        SCRIPT_LANGUAGE: ${{ inputs.script_language }}
+        SCRIPT_NAME: ${{ inputs.script_name }}
+        RESOURCE_GROUP_NAME: ${{ inputs.resource_group_name }}
+        WEB_APP_NAME: ${{ inputs.web_app_name }}
+      run: |
+        set -euo pipefail
+
+        currentPercentage=0
+        stagingSlot=staging
+        prodSlot=production
+
```

It may be nice to show the progress in the github step summary as done for the static analysis (more will follow)
```suggestion
        echo "## 🧪 Incremental Rollout Progress" >> "$GITHUB_STEP_SUMMARY"

```